### PR TITLE
Task/sscs 2810 find ccd case by appeal number

### PR DESCRIPTION
**Pull request by savirneni**
Implement searchCaseByAppealnumber and updateCase functionality for subscription

https://tools.hmcts.net/jira/browse/SSCS-2861

### CODE REVIEW

File: src/e2e/java/uk/gov/hmcts/sscs/ccd/CreateCaseInCcd.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    @Autowired
+    private CreateCoreCaseDataService createCoreCaseDataService;
+
+    @Test
+    public void givenACase_shouldBeSavedIntoCcd() {
+        CaseDetails caseDetails = createCoreCaseDataService.createCcdCase(CaseDataUtils.buildCaseData());
+        assertNotNull(caseDetails);
+    }
+
+}
```
Comment (sathishbabuj) [inline src/e2e/java/uk/gov/hmcts/sscs/ccd/CreateCaseInCcd.java]: Please rename the class to include Test in the name.

File: src/e2e/java/uk/gov/hmcts/sscs/ccd/ReadCaseFromCcd.java [hunk 2]
```diff
@@ -19,8 +19,14 @@
...
+    public void givenACaseId_shouldRetrieveCaseDetails() {
+        CaseDetails caseDetails = readCoreCaseDataService.getCcdCaseDetailsByCaseId("1520966754095462");
+        assertNotNull(caseDetails);
+    }
+
+    @Test
+    public void givenAnAppealNumber_shouldRetrieveCaseDetails() {
+        CaseDetails caseDetails = readCoreCaseDataService.getCcdCaseDetailsByAppealNumber("abcde12345");
         assertNotNull(caseDetails);
     }
```
Comment (sathishbabuj) [inline src/e2e/java/uk/gov/hmcts/sscs/ccd/ReadCaseFromCcd.java]: Please rename the class to include Test in the name.

File: src/main/java/uk/gov/hmcts/sscs/service/ccd/ReadCoreCaseDataService.java [hunk 3]
```diff
@@ -50,15 +52,35 @@ private CaseDetails get(EventRequestData eventRequestData, String serviceAuthori
...
+                                                String appealNumber) {
+        log.info("Get getByAppealNumber...");
+        return coreCaseDataService.getCoreCaseDataApi().searchForCaseworker(
+                eventRequestData.getUserToken(),
+                serviceAuthorization,
+                eventRequestData.getUserId(),
+                eventRequestData.getJurisdictionId(),
+                eventRequestData.getCaseTypeId(),
+                ImmutableMap.of("case.subscriptions.appellantSubscription.tya", appealNumber)
+        );
```
Comment (sathishbabuj) [inline src/main/java/uk/gov/hmcts/sscs/service/ccd/ReadCoreCaseDataService.java]: Log info doesn't contain any useful information. Do we need so many info logs?. Its better to change it to debug and include additional information.

File: src/main/java/uk/gov/hmcts/sscs/service/ccd/UpdateCoreCaseDataService.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                eventRequestData.getUserId(),
+                eventRequestData.getJurisdictionId(),
+                eventRequestData.getCaseTypeId(),
+                caseId.toString(),
+                eventRequestData.isIgnoreWarning(),
+                caseDataContent
+        );
+    }
+
+}
```
Comment (sathishbabuj) [inline src/main/java/uk/gov/hmcts/sscs/service/ccd/UpdateCoreCaseDataService.java]: Too many log info messages

File: src/e2e/java/uk/gov/hmcts/sscs/ccd/CreateCaseInCcd.java (hunk 1 above)
Comment (savirneni) [reply src/e2e/java/uk/gov/hmcts/sscs/ccd/CreateCaseInCcd.java]: done

File: src/e2e/java/uk/gov/hmcts/sscs/ccd/ReadCaseFromCcd.java (hunk 2 above)
Comment (savirneni) [reply src/e2e/java/uk/gov/hmcts/sscs/ccd/ReadCaseFromCcd.java]: done.

File: src/main/java/uk/gov/hmcts/sscs/service/ccd/ReadCoreCaseDataService.java (hunk 3 above)
Comment (savirneni) [reply src/main/java/uk/gov/hmcts/sscs/service/ccd/ReadCoreCaseDataService.java]: Removed the unnecessary info logs.

File: src/main/java/uk/gov/hmcts/sscs/service/ccd/UpdateCoreCaseDataService.java (hunk 4 above)
Comment (savirneni) [reply src/main/java/uk/gov/hmcts/sscs/service/ccd/UpdateCoreCaseDataService.java]: Removed the unnecessary info logs.

**Pull request merged**